src/BenchmarkProfiles.jl

```diff
@@ -1,12 +1,33 @@
 module BenchmarkProfiles
 
-import Plots, NaNMath
+import NaNMath
+using Requires
 using Printf
 
-export performance_ratios, performance_profile
+export performance_ratios, performance_profile, performance_profile_data
 export data_ratios, data_profile
+export bp_backends, PlotsBackend, UnicodePlotsBackend
+
+abstract type AbstractBackend end
+struct PlotsBackend <: AbstractBackend end
+struct UnicodePlotsBackend <: AbstractBackend end
+
+const bp_backends = [:PlotsBackend, :UnicodePlotsBackend]
+
+function throw_error(b)
+  throw(ArgumentError("The backend $b is not loaded. Please load the corresponding package."))
+end
+
+for backend ∈ bp_backends
+  @eval begin
+    B = eval($backend)
+    performance_profile(b::B, ::Matrix{Float64}, ::Vector{S}; kwargs...) where S <: AbstractString = throw_error(b)
+    data_profile(b::B, ::Array{Float64,3}, ::Vector{Float64}, ::Vector{S}; kwargs...) where S <: AbstractString = throw_error(b)
+  end
+end
```

Comment:
```suggestion
function performance_profile(::B, ::Matrix{Float64}, ::Vector{S}; kwargs...) where {B <: AbstractBackend, S <: AbstractString}
  throw(ArgumentError("The backend $B() is not loaded. Please load the corresponding package."))
end
function data_profile(::B, ::Array{Float64,3}, ::Vector{Float64}, ::Vector{S}; kwargs...) where {B <: AbstractBackend, S <: AbstractString}
  throw(ArgumentError("The backend $B() is not loaded. Please load the corresponding package."))
end
```
The `$B()` is to print `PlotsBackend()`, and could be removed.

Reply:
There's a difference here. In my version, the user is alerted that they have to load another package, but only for the backends that have been defined in the package. In your proposal, if I use `UnsupportedBackend` as a backend, I'll be told that loading a package will help, but it won't.

Reply:
But that's assuming a user will define their own `UnsupportedBackend` derived from `BenchmarkProfiles.AbstractBackend` and not define the appropriate function to use it with `BenchmarkProfiles`. If that's what you're trying to avoid, we can resolve this.

Reply:
Julia will naturally throw an error in all other cases.